internal/controllers/watchdog/controller.go

```diff
@@ -62,6 +77,32 @@ func (c *watchdogController) Reconcile(ctx context.Context, req ctrl.Request) (c
 	return ctrl.Result{}, nil
 }
 
+func (c *watchdogController) getSynthesizer(comp *apiv1.Composition, ctx context.Context) (*apiv1.Synthesizer, error) {
+	syn := &apiv1.Synthesizer{}
+	syn.Name = comp.Spec.Synthesizer.Name
+	err := c.client.Get(ctx, client.ObjectKeyFromObject(syn), syn)
+	return syn, err
+}
+
+func (c *watchdogController) hasNoSynthesizer(comp *apiv1.Composition, ctx context.Context) bool {
+	_, err := c.getSynthesizer(comp, ctx)
+	return err != nil
+}
+
+func (c *watchdogController) getInputsExist(comp *apiv1.Composition, ctx context.Context) bool {
+	syn, err := c.getSynthesizer(comp, ctx)
+	return (err != nil) || comp.InputsExist(syn)
```

Comment:
Should this be `err == nil && comp.InputsExist(syn)` to match the logic of getNotInLockstep?

Reply:
No because if `err != nil` then `syn` is not available and from our discussion I got that we treat that condition as if inputs exist, not as if inputs don't exist...